lib/maths/CBoostedTree.cc

```diff
@@ -757,7 +774,7 @@ class CBoostedTree::CImpl final {
         // We allow a large number of trees by default in the main parameter
         // optimisation loop. In practice, we should use many fewer if they
         // don't significantly improve test error.
-        m_MaximumNumberTrees = 500;
+        m_MaximumNumberTrees *= 10;
```

Comment:
would it make sense to bound this to avoid very large numbers?

Reply:
It is implicitly bounded by the value of the learn rate. But I think the observation does then apply to the learn rate override, i.e. we should clearly trap that this is greater than zero. I'll apply some sanity checks on the overrides.